### PR TITLE
Add support for saving and loading scans

### PR DESCRIPTION
**Pull request by Nigecat**

### CODE REVIEW

**Comment by RandomExplosion** [comment]
Crash When Loading Scan With Deleted Files (when the deleted file comes into view)

Steps to reproduce:
  Scan a drive.
  Delete a file on that drive
  Open the scan
  Navigate to that file

**Comment by Nigecat** [comment]
@RandomExplosion that issue should be fixed now in c23dfa2

File: Drive Scan/DriveScan.xaml.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using System.Threading;
+﻿using Microsoft.Win32;
```
Comment (RandomExplosion) [inline Drive Scan/DriveScan.xaml.cs:1]: Woah it doesn't start with System :D

File: Drive Scan/DriveScan.xaml.cs [hunk 2]
```diff
@@ -24,6 +25,8 @@ public partial class DriveScanWindow : MetroWindow
         //Singleton Pattern
         static DriveScanWindow currentWindow;
 
+        static ScanLoader currentScan = new ScanLoader();
```
Comment (RandomExplosion) [inline Drive Scan/DriveScan.xaml.cs:30]: Maybe a comment there?

File: Drive Scan/DriveScan.xaml.cs [hunk 3]
```diff
@@ -233,15 +248,19 @@ public async void ScanDrive(string DriveName)
                 //Show the Bar
                 ProgBar.Visibility = Visibility.Visible;
                 //Scan the drive asynchronously then add the drive tree to the TreeView
-                Task scanTask = Task.Run(() => {Scanning.DirectoryScanner.FindFiles(drive.Name, OnFileFound);
+                Task scanTask = Task.Run(() => {
```
Comment (RandomExplosion) [inline Drive Scan/DriveScan.xaml.cs:253]: Good formatting fix
(my bad :P)

File: Drive Scan/DriveScan.xaml.cs [hunk 4]
```diff
@@ -324,6 +341,61 @@ public void ThemeSwitch(dynamic Sender, RoutedEventArgs e = null)
             }
         }
 
+        /// <Summary>
+        /// Load a scan
+        /// </summary>
+        public void LoadScan(object Sender, RoutedEventArgs e)
```
Comment (RandomExplosion) [inline Drive Scan/DriveScan.xaml.cs:349]: This function still has the comments from when it was copied and pasted from the scanning procedure.
Maybe consider adapting them to the new purpose.

File: Drive Scan/DriveScan.xaml.cs [hunk 5]
```diff
@@ -252,12 +271,10 @@ public async void ScanDrive(string DriveName)
         /// <param name="foundFile"></param>
         /// <param name="isFirstFile"></param>
         /// <param name="isRoot"></param>
-        public void OnFileFound(Scanning.File foundFile, bool isFirstFile, bool isRoot)
+        public void OnFileFound(Scanning.File foundFile)
```
Comment (RandomExplosion) [inline Drive Scan/DriveScan.xaml.cs:276]: Well removed jank

File: Drive Scan/ValueConverters.cs [hunk 6]
```diff
@@ -82,6 +82,8 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
     /// </summary>
     public class AssociatedIconConverter : IValueConverter
     {
+        // Load the error icon as a static variable so we only had to load it once not every time it is needed
+        public static Icon ErrorIcon = Icon.FromHandle((new Bitmap(Image.FromFile(@"Resources\BlankFileError.ico"), new System.Drawing.Size(128, 128))).GetHicon());
```
Comment (RandomExplosion) [inline Drive Scan/ValueConverters.cs:86]: Well Rebranded Bug
Nice Work!
image

**Review by RandomExplosion — CHANGES_REQUESTED** [review]

File: Drive Scan/DriveScan.xaml.cs (hunk 2 above)
Comment (RandomExplosion) [reply Drive Scan/DriveScan.xaml.cs:30]: Just so we know what it does

**Review by RandomExplosion — APPROVED** [review]

**Pull request merged**